patches/minecraft/net/minecraft/world/item/ShieldItem.java.patch

```diff
@@ -0,0 +1,14 @@
+--- a/net/minecraft/world/item/ShieldItem.java
++++ b/net/minecraft/world/item/ShieldItem.java
+@@ -49,4 +_,11 @@
+    public static DyeColor m_43102_(ItemStack p_43103_) {
+       return DyeColor.m_41053_(p_43103_.m_41698_("BlockEntityTag").m_128451_("Base"));
+    }
++
++   /* ******************** FORGE START ******************** */
++
++   @Override
++   public boolean canPerformAction(ItemStack stack, net.minecraftforge.common.ToolAction toolAction) {
++      return net.minecraftforge.common.ToolActions.DEFAULT_SHIELD_ACTIONS.contains(toolAction);
++   }
```

Comment:
Why add this patch here instead of doing what `isShield` used to do in `canPerformAction` (or rather, just call `isShield` which already checks this...)?

Reply:
First, it is consistent with the pre-existing implementation of `ToolAction`. If I look into `SwordItem`, I see `canPerformAction` overriden. If I look into `PickaxeItem`, I see `canPerformAction` overriden. Same with `ShovelItem`, `AxeItem`, `HoeItem`, and `ShearsItem`. Leaving `ShieldItem` out of this consistency will make it the odd one out, and it will seem odd why shields are given that specific treatment.

Second, it will be easier for the next person who comes to remove `isShield` to do so. They only have to do two things now: remove `isShield` from both `IForgeItem` and `IForgeItemStack`, and change the implementation of `IForgeItem#canPerformAction` to `return false`. Doing the former will cause a compile error in the latter, and the comment in the method body will steer them to change it to `return false`. This also avoids changing the patch later.

Third, if we move the check which exists currently in `isShield` to `IForgeItem#canPerformAction`, then any mod currently made which uses `isShield` in their custom shield item will not work, as then `isShield` would never be called. As I've seen Lex state that Forge 1.17 is effectively API-stable, I'd like to avoid breaking `isShield`. Furthermore, `IForgeItem#canPerformAction` already calls `IForgeItem#isShield`.

Reply:
Fair enough